libisl: Make derivations generic, add 0.24

matthiasbeyer opened the pull request:
###### Motivation for this change

Nothing in particular

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


Could not build because my infra is not powerful enough to build this (my attempt tried to rebuild gcc10...)

matthiasbeyer committed:
libisl: Make derivation generic

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
libisl: add 0.24

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

r-burns commented on pkgs/development/libraries/isl/0.17.1.nix:
```diff
@@ -1,24 +1,11 @@
 { lib, stdenv, fetchurl, gmp }:
```
```suggestion
{ callPackage, lib, stdenv, fetchurl, gmp }:
```

Some of the older versions are missing this callPackage arg

matthiasbeyer replied:
whoops, that slipped. Thanks!

matthiasbeyer commented:
I will rebase as soon as somebody approves this :smile:

r-burns commented on pkgs/development/libraries/isl/0.24.nix:
```diff
@@ -0,0 +1,16 @@
+{ callPackage, lib, stdenv, fetchurl, gmp }:
```
I think this file needs to be named 0.24.0.nix

r-burns commented on pkgs/development/libraries/isl/0.20.0.nix:
```diff
@@ -1,29 +1,16 @@
+callPackage ./generic.nix rec {
+  version = "0.20";
+  urls = [
+    "mirror://sourceforge/libisl/libisl-${version}.tar.xz"
+    "https://libisl.sourceforge.io/libisl-${version}.tar.xz"
```
```suggestion
    "mirror://sourceforge/libisl/isl-${version}.tar.xz"
    "https://libisl.sourceforge.io/isl-${version}.tar.xz"
```

same goes for the others, I think

r-burns commented:
The callPackage-inside-callPackage pattern here breaks overriding so stdenv won't build. Switching to `import ./generic.nix` fixes this. The changes are too big for an inline comment so I'll issue a PR against your fork to demonstrate.

r-burns committed:
isl: remove unused args

r-burns committed:
isl: remove dead homepages

r-burns committed:
isl: fix configureFlags

r-burns committed:
isl: fix nested callPackage

use import to remove callPackage-inside-callPackage, which breaks overriding

r-burns committed:
isl: pname/version -> name

pname/version is preferred, but causes rebuilds here.
let's do the cleanup separately from the mass rebuild,
to use the 0-rebuild-count as a 'nothing up my sleeve' gesture

r-burns committed:
isl_0_24: fix sha256

r-burns committed:
isl_0_11: add missing patch

matthiasbeyer commented:
Merged your patches and rebase (+squashed my fixups) the branch on master. :tada:

r-burns commented:
Awesome, exactly 1 rebuild so everything worked :)

Did you want keep `make derivation generic` and `add 0.24` as two separate commits? Otherwise I'll just squash-and-merge.

matthiasbeyer commented:
I would merge it as-is, without any squashing because I think squashing just destroys history! :smiley:

r-burns commented:
heh, well for a normal project I'd agree, but I think we're trying to squash when possible to keep the git history from growing out of control

matthiasbeyer commented:
Well then, I'd say keep two separate commits for the generic making and one for the new version...

But you can squash your commits into mine if you want to.

Either way, I don't see any point in 


> keep the git history from growing out of control

But that should be discussed elsewhere 😂

r-burns commented:
Fair enough, to be honest I didn't make that rule, I just don't want to get yelled at for bloating the history ;)

Anyway, this LGTM :)

matthiasbeyer commented:
I think the "push to master" people bloat the history way more! ... I've been yelled at for that statement several times, yet I am not tired of making it again and again :laughing:
